bin/minifollowups/pycbc_plot_trigger_timeseries

```diff
@@ -49,70 +51,70 @@ pycbc.init_logging(args.verbose)
-    # center times on the trigger/chosen time
-    times = times - t
-
-    if args.plot_type == 'snr':
-        val = snr
-
-        if args.special_trigger_ids:
-            top = data[ifo]['snr'][id_loud]
-
-    if args.plot_type == 'newsnr':
-        rchisq = chisq / (2 * chisq_dof - 2)
-        if len(rchisq) > 0:
-            val = ranking.newsnr(snr, rchisq)
-        else:
-            val = numpy.array([])
-
-        # Get the newnsr of the loudest trigger so we can plot a star there
-        if args.special_trigger_ids:
-            rchisq = data[ifo]['chisq'][id_loud] / \
-                (data[ifo]['chisq_dof'][id_loud] * 2 - 2)
-            top = ranking.newsnr(data[ifo]['snr'][id_loud], rchisq)
-
-    if type(val) in [numpy.float32, numpy.float64] or len(val) > 0:
-        any_data = True
-
-    pylab.scatter(times, val, color=pycbc.results.ifo_color(ifo), marker='x',
+    keep = abs(data[ifo + '/end_time'][:] - t) < args.window
+    if not any(keep):
+        # No triggers in this window, add to the legend and continue
+        pylab.scatter(0, 0, color=pycbc.results.ifo_color(ifo), marker='x',
+                  label=ifo)
+        continue
+
+    any_data = True
+    logging.info("Keeping %d triggers in the window", sum(keep))
+
+
+    data_dict = {k: data[ifo][k][keep] for k in data[ifo].keys()
+                 if isinstance(data[ifo][k], h5py.Dataset)
+                 and data[ifo][k].size == keep.size}
+
+    logging.info("Getting %s", args.plot_type)
+    rank = ranking.get_sngls_ranking_from_trigs(data_dict, args.plot_type)
+
+    if all(rank == 1):
+        # This is the default value to say "downranked beyond consideration"
+        # so skip these events
+        pylab.scatter(0, 0, color=pycbc.results.ifo_color(ifo), marker='x',
+                  label=ifo)
+        continue
+
+    pylab.scatter(data_dict['end_time'] - t, rank,
+                  color=pycbc.results.ifo_color(ifo), marker='x',
                   label=ifo)
 
+    # Minimum rank which hasn't been set to the default of 1
+    min_rank = min(min_rank, rank[rank > 1].min())
+
     if args.special_trigger_ids:
-        any_data = True
-        pylab.scatter([0], [top], marker='*', s=50, color='yellow')
+        special_idx = numpy.flatnonzero(keep) == args.special_trigger_ids[ifo]
+        if not any(special_idx):
+            logging.info("IDX %d not in kept list", args.special_trigger_ids[ifo])
+            continue
+
+        pylab.scatter((data_dict['end_time'] - t)[special_idx],
+                      rank[special_idx], marker='*', s=50, color='yellow')
 
 if args.log_y_axis and any_data:
     pylab.yscale('log')
 
+if not numpy.isinf(min_rank):
+    pylab.ylim(ymin=min_rank)
+
 pylab.xlabel('time (s)')
 pylab.ylabel(args.plot_type)
 
-try:
-    if len(val) > 0:
-        pylab.ylim(ymin=val.min())
-except TypeError:
-    pylab.ylim(ymin=val)
-
 pylab.xlim(xmin=-args.window, xmax=args.window)
 pylab.legend()
 pylab.grid()
+
```

Comment:
Can we add a `logging.info("Saving figure")` here and a `logging.info("Done")` at the very end, just to debug cases where things get unexplicably slow?